server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -271,6 +271,8 @@ public void verifyEngineBeforeIndexClosing() throws IllegalStateException {
         }
     }
 
+    public void finalizeReplication(SegmentInfos infos, Store.MetadataSnapshot expectedMetadata, long seqNo) throws IOException {}
```

Comment:
This method already exists within NRTEngine as `updateSegments`.  I don't think we need to define it in Engine as we'll want to assert & cast the engine type when finalize is called on the shard.